setup.py

```diff
@@ -1,163 +1,223 @@
-            print("Couldn't install PyOD for testing!")
-            try_run_setup(**kwargs)
-        else:
-            print("ERROR: Failed to build!")
-
-# we seem to need this import guard for appveyor
-if __name__ == "__main__":
-    try_run_setup(with_binary=True, test_xgboost=True, test_lightgbm=True, test_spark=True, test_pyod=True)
+from setuptools import setup, Extension
+from setuptools.command.build_ext import build_ext as _build_ext
+import os
+import re
+import codecs
+import platform
+from distutils.sysconfig import get_config_var, get_python_inc
+from distutils.version import LooseVersion
+import sys
+import subprocess
+
+# to publish use:
+# > python setup.py sdist bdist_wheel upload
+# which depends on ~/.pypirc
+
+
+# This is copied from @robbuckley's fix for Panda's
+# For mac, ensure extensions are built for macos 10.9 when compiling on a
+# 10.9 system or above, overriding distuitls behavior which is to target
+# the version that python was built for. This may be overridden by setting
+# MACOSX_DEPLOYMENT_TARGET before calling setup.py
+if sys.platform == 'darwin':
+    if 'MACOSX_DEPLOYMENT_TARGET' not in os.environ:
+        current_system = LooseVersion(platform.mac_ver()[0])
+        python_target = LooseVersion(get_config_var('MACOSX_DEPLOYMENT_TARGET'))
+        if python_target < '10.9' and current_system >= '10.9':
+            os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
+
+here = os.path.abspath(os.path.dirname(__file__))
+
+
+def read(*parts):
+    with codecs.open(os.path.join(here, *parts), 'r') as fp:
+        return fp.read()
+
+
+def find_version(*file_paths):
+    version_file = read(*file_paths)
+    version_match = re.search(r"^__version__ = ['\"]([^'\"]*)['\"]", version_file, re.M)
+    if version_match:
+        return version_match.group(1)
+    raise RuntimeError("Unable to find version string.")
+
+
+# Extend the default build_ext class to bootstrap numpy installation
+# that are needed to build C extensions.
+# see https://stackoverflow.com/questions/19919905/how-to-bootstrap-numpy-installation-in-setup-py
+class build_ext(_build_ext):
+    def finalize_options(self):
+        _build_ext.finalize_options(self)
+        if isinstance(__builtins__, dict):
+            __builtins__["__NUMPY_SETUP__"] = False
+        else:
+            setattr(__builtins__, "__NUMPY_SETUP__", False)
+        import numpy
+        print("numpy.get_include()", numpy.get_include())
+        self.include_dirs.append(numpy.get_include())
+
+
+def compile_cuda_module(host_args):
+    libname = '_cext_gpu.lib' if sys.platform == 'win32' else 'lib_cext_gpu.a'
+    lib_out = 'build/' + libname
+    nvcc = os.path.join(os.path.abspath(os.environ['CUDA_PATH']), 'bin/nvcc')
+    arch_flags = "-arch=sm_60 " + \
+                 "-gencode=arch=compute_70,code=sm_70 " + \
+                 "-gencode=arch=compute_75,code=sm_75 " + \
+                 "-gencode=arch=compute_75,code=compute_75"

```

Comment:
I would add 80 if possible (Ampere), that would cover the bulk of GPU architectures probably

Reply:
I was trying to avoid discovering the cuda toolkit version for now, as I would have to do this manually and 80 presumably won't work on cuda 10.1.

Reply:
ah totally true, I blanked for a sec about that requirement, then I think this is the best choice.